src/OrchardCore/OrchardCore.DisplayManagement/Razor/RazorPage.cs

```diff
@@ -51,6 +54,14 @@ private void EnsureShapeFactory()
             }
         }
 
+        private void EnsureShellFeaturesManager()
```

Comment:
I think this method would be better if it returned the cached instance if `IShellFeaturesManager` and maybe name name `GetFeatureManager()`

Reply:
You can suggest this into a new issue. I'm not sure if we still need this PR ;)